packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -100,20 +102,49 @@ class ArtifactBuilder {
         ...args,
       ];
 
-      final result = await process.run(
+      final buildProcess = await process.start(
         executable,
         arguments,
         runInShell: true,
         environment: base64PublicKey?.toPublicKeyEnv(),
       );
 
-      if (result.exitCode != ExitCode.success.code) {
-        throw ArtifactBuildException(
-          'Failed to build: ${result.stderr}',
-        );
+      // Android builds are a series of gradle tasks that are all logged in
+      // this format. We can use the 'Task :' line to get the current task
+      // being run.
+      final gradleTaskRegex = RegExp(r'^\[.*\] \> (Task :.*)$');
+      buildProcess.stdout
+          .transform(utf8.decoder)
+          .transform(const LineSplitter())
+          .listen((line) {
+        if (buildProgress == null) {
+          return;
+        }
+        final match = gradleTaskRegex.firstMatch(line);
+        if (match != null) {
+          final captured = match.group(1);
+          if (captured != null) {
+            buildProgress.updateDetailMessage(captured);
+          }
+        }
+      });
+
+      final stderrLines = await buildProcess.stderr
+          .transform(utf8.decoder)
+          .transform(const LineSplitter())
+          .toList();
+      final stdErr = stderrLines.join('\n');
+      final exitCode = await buildProcess.exitCode;
+      if (exitCode != ExitCode.success.code) {
+        throw ArtifactBuildException('Failed to build: $stdErr');
       }
     });
 
+    // If we've been updating the progress with gradle tasks, reset it to the
+    // original base message so as not to leave the user with a confusing
+    // message.
+    buildProgress?.updateDetailMessage(null);
```

Comment:
I guess this is a setter?  buildProgress?.detailMessage = null;

Reply:
It could be. I used `updateXxxxMessage` methods for consistency with the `update` function.